primary_beam_correction.py

```diff
@@ -178,136 +151,119 @@ def standard_deviation(data):
-    corr_image = corr_image[np.newaxis,np.newaxis,:,:]
+    corr_image = corr_image.reshape(true_image.data.shape[2], true_image.data.shape[3])
+    # Add new axis
+    corr_image = corr_image[np.newaxis, np.newaxis, :, :]
     return corr_image
-                   
+
+
 def write_new_fits(pbc_image, path, outputFilename):
     '''
     Write out a new fits image with primary beam corrected continuum in its first  plane
     '''
     images = read_fits(path)
-    
     hdr = images.header
     newhdr = hdr.copy()
     # add in required beam keywords
-    # get the data and select only the first frequency plane
-    try:
-        if newhdr['CLEANBMJ']< 0:
-            # add in required beam keywords
-            newhdr['BMAJ'] = float(newhdr['HISTORY'][-2][20:30])
-            newhdr['BMIN'] = float(newhdr['HISTORY'][-2][36:48])
-            newhdr['BPA']  = float(newhdr['HISTORY'][-2][55:])
+    if newhdr['CLEANBMJ'] < 0:
+        # add in required beam keywords
+        newhdr['BMAJ'] = float(newhdr['HISTORY'][-2][20:30])
+        newhdr['BMIN'] = float(newhdr['HISTORY'][-2][36:48])
+        newhdr['BPA'] = float(newhdr['HISTORY'][-2][55:])
 
-            newhdr['CTYPE3']='FREQ'
-            newhdr['NAXIS3']=1
-            newhdr['CDELT3']=1.0
-            newhdr['CRVAL3']= 1.284e9
-        else:
-            # add in required beam keywords
-            newhdr['BMAJ'] = newhdr['CLEANBMJ']
-            newhdr['BMIN'] = newhdr['CLEANBMN']
-            newhdr['BPA'] = newhdr['CLEANBPA']
-            #change the frequency plane keywords, we don't want multiple frequency #axes
-            newhdr['CTYPE3']='FREQ'
-            newhdr['NAXIS3']=1
-            newhdr['CDELT3']=1.0
-            newhdr['CRVAL3']= 1.284e9
-    except:
-        newhdr['BMAJ'] = float(newhdr['HISTORY'][5][18:30])
-        newhdr['BMIN'] = float(newhdr['HISTORY'][7][18:30])
-        newhdr['BPA']  = float(newhdr['HISTORY'][9][18:30])
-        newhdr['CTYPE3']='FREQ'
-        newhdr['NAXIS3']=1
-        newhdr['CDELT3']=1.0
-        newhdr['CRVAL3']= 1.284e9
-        
-    data_new = pbc_image[:,0:1,:,:] # the shape of data_new = (1,1, npix, npix)
+        newhdr['CTYPE3'] = 'FREQ'
+        newhdr['NAXIS3'] = 1
+        newhdr['CDELT3'] = 1.0
+        newhdr['CRVAL3'] = 1.284e9
+    else:
+        # add in required beam keywords
+        newhdr['BMAJ'] = newhdr['CLEANBMJ']
+        newhdr['BMIN'] = newhdr['CLEANBMN']
+        newhdr['BPA'] = newhdr['CLEANBPA']
+        # change the frequency plane keywords, we don't want multiple frequency
+        newhdr['CTYPE3'] = 'FREQ'
+        newhdr['NAXIS3'] = 1
+        newhdr['CDELT3'] = 1.0
+
+    data_new = pbc_image[:, 0:1, :, :]  # the shape of data_new = (1,1, npix, npix)
     new_hdu = fits.PrimaryHDU(header=newhdr, data=data_new)
-    
     new_hdu.writeto(outputFilename)
-                   
-if __name__=="__main__":
-    args = docopt(__doc__)
-    path = args['--path'] 
+    return
 
+
+if __name__ == "__main__":
```

Comment:
Better to create a function:
`def main(): .....`
and put everything inside here into that function then the script level just contains:
```
if __name__ == "__main__":
    main()
```

This means you can import the script in other python code and run it away from the command line.

Reply:
Done.

Reply:
Done.